p5js/PoseNet/sketch.js

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2018 ml5
+// 
+// This software is released under the MIT License.
+// https://opensource.org/licenses/MIT
+
+/* ===
+ML5 Example
+PoseNet using p5.js
+=== */
+
+let w = 315;
+let h = 315;
+let video;
+let poseNet;
+let poses = [];
+let skeletons = [];
+
+function setup() {
+  createCanvas(w, h);
+  video = createCapture(VIDEO);
+  
+  // Create a new poseNet method with a single detection
+  poseNet = new ml5.PoseNet(video, 'single', gotPoses);
+  
+  // Hide the video element, and just show the canvas
+  video.hide();
+  fill(255, 0, 0);
+  stroke(255, 0, 0);
+}
+
+function draw() {
+  image(video, 0, 0, w, h);
+
+  // We can call both functions to draw all keypoints and the skeletons
+  drawKeypoints();
+  drawSkeleton();
+}
+
+// A function to draw ellipses over the detected keypoints
+function drawKeypoints() {
+  poses.forEach(pose => {
+    pose.keypoints.forEach(e => {
+      if (e.score > 0.2) {
+        ellipse(e.position.x, e.position.y, 10, 10);
+      }
+    })
+  });
+}
+
+// A function to draw the skeletons
+function drawSkeleton() {
+  skeletons.forEach(skeleton => {
+    skeleton.forEach(parts => {
+      let x1 = parts[0].position.x;
+      let y1 = parts[0].position.y;
+      let x2 = parts[1].position.x;
+      let y2 = parts[1].position.y;
+      line(x1, y1, x2, y2);
+    });
+  });
+}
+
+// The callback that gets called every time there's an update from the model
+function gotPoses(results) {
+  poses = results;
+  skeletons = poses.map(pose => poseNet.skeleton(pose.keypoints));
```

Comment:
Is this something we can package as a feature of the library? 

```javascript
function gotPoses(poses) {
 skeletons = poseNet.skeleton(poses);
}
```

or

```javascript
function gotPoses(poses, skeletons) {
 // callback passes arrays of both?
 // Does the end user always want the "skeleton" format?
}
```

At the very least we might want to add some comments / explanation of what's happening here.

Reply:
totally! I think the later is better, will update then

Reply:
I'm a little confused as to the difference between a `pose` and a `skeleton` but I can take a look at the example and play with it a bit more after we merge!

Reply:
`pose` is an array of keypoints describing the body positions detected. 
```javascript
[ score: 0.9952507019042969, part: "nose", position: { x: 174.63247075574154, y: 226.84211836190062 } , ...]
```

`skeleton` is an array containing arrays of two elements describing connections between two keypoints
```javascript
[[part: "leftShoulder"], [ part: "rightShoulder" ], ... ]
```

Reply:
Ah, that makes sense! Let me know when you are ready for me to merge!